runtime/ops/fs.rs

```diff
@@ -187,9 +191,9 @@ fn op_open_sync(
 ) -> Result<ResourceId, AnyError> {
   let (path, open_options) =
     open_helper(state, &path, mode, options.as_ref(), "Deno.openSync()")?;
-  let std_file = open_options.open(&path).map_err(|err| {
-    Error::new(err.kind(), format!("{}, open '{}'", err, path.display()))
-  })?;
+  let std_file = open_options
+    .open(&path)
+    .map_err(create_err_mapper(format!("open '{}'", path.display())))?;
```

Comment:
This will always allocate a string for error even if error is not returned. Could you instead change these to use closure instead so they are lazily evaluated?

Reply:
On the one hand, this looks ugly as hell:

```
.map_err(|err| create_err_mapper(format!("open '{}'", path.display()))(err))?;
```

But this one isn't any better as well :(

```
fn create_err_mapper(desc: impl Fn() -> String) -> impl Fn(Error) -> Error {
  move |err: Error| Error::new(err.kind(), format!("{err}, {}", desc()))
}

// ...

.map_err(create_err_mapper(|| format!("open '{}'", path.display())))?;
```

Any preferences?

Reply:
The first one would be fairly pretty if it were not for the `create_err_mapper` usage which here seems a bit redundant as it's being created inline in a lambda anyway.

Would it be possible to write a secondary `err_mapper` function for use in these "inline" cases when there is no real benefit in creating an error mapper once and using it in many places?

```rs
.map_err(|err| default_err_mapper(format!("open '{}'", path.display()), err))
```